{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "240a3ae6-c942-4cf5-b62a-09b19590f749",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "df1=pd.read_csv('athlete_events.csv')\n",
    "df2=pd.read_csv('noc_regions.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a05c8969-6ccf-404c-a3cb-0a5db9ebd4eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# USER_NAME , PASSWORD , PORT , DATAbane name \n",
    "\n",
    "from sqlalchemy import create_engine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d0f0aa81-add2-4503-98f5-7799cc16e0eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "connection_string = \"mysql+pymysql://root:python123@localhost:3306/olympics\"\n",
    "engine = create_engine(connection_string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d089719d-52f1-4ffb-91ca-b8c7c1833d2c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\HP\\AppData\\Local\\Temp\\ipykernel_9644\\1428016684.py:1: UserWarning: The provided table name 'OLYMPICS_HISTORY' is not found exactly as such in the database after writing the table, possibly due to case sensitivity issues. Consider using lower case table names.\n",
      "  df1.to_sql('OLYMPICS_HISTORY', con=engine, if_exists='append', index=False)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " sucessfully insert the data into the dtaabases \n"
     ]
    }
   ],
   "source": [
    "df1.to_sql('OLYMPICS_HISTORY', con=engine, if_exists='append', index=False)\n",
    "\n",
    "print(\" sucessfully insert the data into the dtaabases \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "892ef3bd-a45b-47af-8aba-2976305979e2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\HP\\AppData\\Local\\Temp\\ipykernel_9644\\4178712590.py:1: UserWarning: The provided table name 'OLYMPICS_HISTORY_NOC_REGIONS' is not found exactly as such in the database after writing the table, possibly due to case sensitivity issues. Consider using lower case table names.\n",
      "  df2.to_sql('OLYMPICS_HISTORY_NOC_REGIONS', con=engine, if_exists='append', index=False)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "230"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2.to_sql('OLYMPICS_HISTORY_NOC_REGIONS', con=engine, if_exists='append', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "aeb133e5-c91d-4a73-a477-76cbad92742f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(271116, 15)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de3d9260-5f33-4c69-a596-6ad4266cfac2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
